OpenEdX/AppDelegate.swift

```diff
@@ -62,33 +63,44 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
             object: nil
         )
         
+        if let pushManager = Container.shared.resolve(PushNotificationsManager.self) {
+            pushManager.performRegistration()
+        }
+
         return true
     }
 
     func application(
         _ app: UIApplication,
         open url: URL, options: [UIApplication.OpenURLOptionsKey: Any] = [:]
     ) -> Bool {
-        if let config = Container.shared.resolve(ConfigProtocol.self) {
-            if config.facebook.enabled {
-                ApplicationDelegate.shared.application(
-                    app,
-                    open: url,
-                    sourceApplication: options[UIApplication.OpenURLOptionsKey.sourceApplication] as? String,
-                    annotation: options[UIApplication.OpenURLOptionsKey.annotation]
-                )
-            }
+        guard let config = Container.shared.resolve(ConfigProtocol.self) else { return false }
 
-            if config.google.enabled {
-                return GIDSignIn.sharedInstance.handle(url)
+        if let deepLinkManager = Container.shared.resolve(DeepLinkManager.self),
+            deepLinkManager.anyServiceEnabled {
+            if deepLinkManager.handledURLWith(app: app, open: url, options: options) {
+                return true
             }
+        }
 
-            if config.microsoft.enabled {
-                return MSALPublicClientApplication.handleMSALResponse(
-                    url,
-                    sourceApplication: options[UIApplication.OpenURLOptionsKey.sourceApplication] as? String
-                )
-            }
+        if config.facebook.enabled {
+            ApplicationDelegate.shared.application(
+                app,
+                open: url,
+                sourceApplication: options[UIApplication.OpenURLOptionsKey.sourceApplication] as? String,
+                annotation: options[UIApplication.OpenURLOptionsKey.annotation]
+            )
+        }
```

Comment:
Although it's not part of the underlying PR, but as we are, how about fixing it as well, it can return false, whereas the method should have been called for Google or Microsoft. It would be nice if we see if the URL is for Facebook only then it will be returning the status.